File: frontend/src/types/student.ts
/*
 * FEATURE CODE: QUAN-20260529-0943
 * Project: ONENET Student Management System
 * Layer: Frontend (TypeScript Types)
 */

export interface Student {
  id: string;
  studentCode: string;
  fullName: string;
  dateOfBirth: string; // ISO format or YYYY-MM-DD
  gender: 'Nam' | 'Nữ' | 'Khác';
  address?: string;
  parentPhone: string;
  email?: string;
  createdAt?: string;
}

export interface Pagination {
  currentPage: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
}

export interface StudentsResponse {
  success: boolean;
  data: Student[];
  pagination: Pagination;
}

File: src/ONENET.Domain/Entities/Student.cs
/*
 * FEATURE CODE: QUAN-20260529-0943
 * Project: ONENET Student Management System
 * Layer: Domain (Entities)
 */

using System;

namespace ONENET.Domain.Entities
{
    public class Student
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string StudentCode { get; set; } = string.Empty;
        public string FullName { get; set; } = string.Empty;
        public DateOnly DateOfBirth { get; set; }
        public string Gender { get; set; } = string.Empty;
        public string? Address { get; set; }
        public string ParentPhone { get; set; } = string.Empty;
        public string? Email { get; set; }
        public bool IsDeleted { get; set; } = false;
        public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
        public DateTimeOffset UpdatedAt { get; set; } = DateTimeOffset.UtcNow;
    }
}